paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ach/USBankAccountFormScreenState.kt

```diff
@@ -13,7 +13,7 @@ internal sealed class USBankAccountFormScreenState(
     abstract val mandateText: String?
 
     @Parcelize
-    data class BillingDetailsCollection(
+    class BillingDetailsCollection(
```

Comment:
This is the one line fix. data class always made this screen state update equal, therefore never updating the primary button state.

Reply:
This doesn't feel like the right fix.

Can you explain this more? (happy to jump on a call as well).

Reply:
+1 on this. Is the issue that `handleScreenStateChanged()` isn’t invoked after canceling the flow?